x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/DestinationIndex.java

```diff
@@ -303,6 +308,18 @@ public static void updateMappingsToDestIndex(Client client,
         getFieldCapsForRequiredFields(client, config, fieldCapabilitiesListener);
     }
 
+    private static ImmutableOpenMap<String, MappingMetadata> createEmptyMappings() {
+        MappingMetadata singleDocEmptyMetadata;
+        try {
+            singleDocEmptyMetadata = new MappingMetadata(MapperService.SINGLE_MAPPING_NAME, Collections.emptyMap());
+        } catch (IOException e) {
+            throw new IllegalStateException("we should never fail to create empty mapping_metadata");
```

Comment:
```suggestion
            // we should never fail to create empty mapping_metadata
            throw new IllegalStateException("failed to create empty mapping_metadata");
```
Since this exception MAY (though should never) be written to a log read by the user.